api/api/controllers/search_controller.py

```diff
@@ -597,7 +620,13 @@ def get_sources(index):
         except NotFoundError:
             buckets = [{"key": "none_found", "doc_count": 0}]
         sources = {result["key"]: result["doc_count"] for result in buckets}
-        cache.set(key=source_cache_name, timeout=SOURCE_CACHE_TIMEOUT, value=sources)
+        if is_redis_reachable:
+            cache.set(
+                key=source_cache_name, timeout=SOURCE_CACHE_TIMEOUT, value=sources
+            )
+        else:
+            log.warning("Redis connect failed, cannot cache sources.")
```

Comment:
It does say the opposite thing, one was for being unable to read from the cache and the other is for being unable to write to it. The `is_redis_reachable` variable is just to avoid a potential network call to a service that's not accepting connections.

Reply:
Right, but in the context of reading the logs and trying to understand what happened, we'd already know it didn't write to the sources cache because it wasn't able to read them, right?

I actually wonder if it's the right call to have this cached `is_redis_unreachable` approach. Theoretically it could become reachable between the first and second call, right? Otherwise, why not have a context variable in the entire request that checks if Redis is reachable once and for all, then bypasses any requests if not?

Reply:
In theory it could, I'll update the calls to be independent (here as well as in the other place).

Reply:
Fixed in 65f7dba.